From 7c10b832c12f6be6b59bfd1e378f218f22058f56 Mon Sep 17 00:00:00 2001
From: Rabeeh Khoury <rabeeh@solid-run.com>
Date: Mon, 23 Dec 2013 12:17:43 +0200
Subject: [PATCH] PCI-E is per board and not general to microsom

CuBox-i doesn't have pci-e; so the initialization must be outside the microsom and into the board specific things

Signed-off-by: Rabeeh Khoury <rabeeh@solid-run.com>
---
 arch/arm/mach-mx6/board-mx6q_c1.c       | 10 +++++++++-
 arch/arm/mach-mx6/board-mx6q_microsom.c |  9 ---------
 2 files changed, 9 insertions(+), 10 deletions(-)

diff --git a/arch/arm/mach-mx6/board-mx6q_c1.c b/arch/arm/mach-mx6/board-mx6q_c1.c
index 081daaa..9f515aa 100644
--- a/arch/arm/mach-mx6/board-mx6q_c1.c
+++ b/arch/arm/mach-mx6/board-mx6q_c1.c
@@ -115,6 +115,13 @@
 	},
 };
 
+static const struct imx_pcie_platform_data mx6_c1_pcie_data __initconst = {
+	.pcie_pwr_en	= -EINVAL,
+	.pcie_rst	= -EINVAL,
+	.pcie_wake_up	= -EINVAL,
+	.pcie_dis	= -EINVAL,
+};
+
 static void imx6q_c1_usbotg_vbus(bool on)
 {
 	if (on)
@@ -364,7 +371,8 @@ static void __init mx6_c1_board_init(void)
 	imx6q_add_spdif(&mxc_spdif_data);
 	imx6q_add_spdif_dai();
 	imx6q_add_spdif_audio_device();
-
+	/* Add PCIe RC interface support */
+	imx6q_add_pcie(&mx6_c1_pcie_data);
 #ifdef CONFIG_IR_GPIO_CIR
 	/* Register the infra red receiver as a GPIO device */
 	platform_device_register(&c1_ir);
diff --git a/arch/arm/mach-mx6/board-mx6q_microsom.c b/arch/arm/mach-mx6/board-mx6q_microsom.c
index c5153bf..f458d74 100644
--- a/arch/arm/mach-mx6/board-mx6q_microsom.c
+++ b/arch/arm/mach-mx6/board-mx6q_microsom.c
@@ -542,13 +542,6 @@ static int __init caam_setup(char *__unused)
 }
 early_param("caam", caam_setup);
 
-static const struct imx_pcie_platform_data mx6_usom_pcie_data __initconst = {
-	.pcie_pwr_en	= -EINVAL,
-	.pcie_rst	= -EINVAL,
-	.pcie_wake_up	= -EINVAL,
-	.pcie_dis	= -EINVAL,
-};
-
 static int usom_bt_power_change(int status)
 {
 	if(status){
@@ -754,8 +747,6 @@ void __init mx6_usom_board_init(void)
 
 	imx6q_add_busfreq();
 
-	/* Add PCIe RC interface support */
-	imx6q_add_pcie(&mx6_usom_pcie_data);
 	imx6_add_armpmu();
 	imx6q_add_perfmon(0);
 	imx6q_add_perfmon(1);
-- 
1.8.5.1

